eventsourcing/src/main/java/org/axonframework/eventsourcing/MultiStreamableMessageSource.java

```diff
@@ -0,0 +1,476 @@
+            if (messageSourceMap.containsKey(longPollingSource)) {
+                sourceList.add(new IdentifiedStreamableMessageSource(longPollingSource, messageSourceMap.get(longPollingSource)));
+            }
+            return sourceList;
+        }
+    }
+
+    private static class IdentifiedStreamableMessageSource implements StreamableMessageSource<TrackedEventMessage<?>> {
+
+        private final StreamableMessageSource<TrackedEventMessage<?>> delegate;
+        private final String sourceId;
+
+        public IdentifiedStreamableMessageSource(String sourceId, StreamableMessageSource<TrackedEventMessage<?>> delegate) {
+            this.delegate = delegate;
+            this.sourceId = sourceId;
+        }
+
+        @Override
+        public BlockingStream<TrackedEventMessage<?>> openStream(TrackingToken trackingToken) {
+            return delegate.openStream(trackingToken);
+        }
+
+        @Override
+        public TrackingToken createTailToken() {
+            return delegate.createTailToken();
+        }
+
+        @Override
+        public TrackingToken createHeadToken() {
+            return delegate.createHeadToken();
+        }
+
+        @Override
+        public TrackingToken createTokenAt(Instant dateTime) {
+            return delegate.createTokenAt(dateTime);
+        }
+
+        @Override
+        public TrackingToken createTokenSince(Duration duration) {
+            return delegate.createTokenSince(duration);
+        }
+
+        public String sourceId() {
+            return sourceId;
+        }
+    }
+
+    private static class SourceIdAwareBlockingStream implements BlockingStream<TrackedEventMessage<?>> {
+
+        private final String sourceId;
+        private final BlockingStream<TrackedEventMessage<?>> delegate;
+
+        public SourceIdAwareBlockingStream(String sourceId, BlockingStream<TrackedEventMessage<?>> delegate) {
+            this.sourceId = sourceId;
+            this.delegate = delegate;
+        }
+
+        @Override
+        public boolean hasNextAvailable() {
+            return delegate.hasNextAvailable();
+        }
+
+        @Override
+        public Optional<TrackedEventMessage<?>> peek() {
+            return delegate.peek();
+        }
+
+        @Override
+        public boolean hasNextAvailable(int timeout, TimeUnit unit) throws InterruptedException {
+            return delegate.hasNextAvailable(timeout, unit);
+        }
+
+        @Override
+        public TrackedEventMessage<?> nextAvailable() throws InterruptedException {
+            return delegate.nextAvailable();
+        }
+
+        @Override
+        public void close() {
+            delegate.close();
+        }
+
+        @Override
+        public Stream<TrackedEventMessage<?>> asStream() {
+            return delegate.asStream();
+        }
+    }
+
+    private static class MultiSourceBlockingStream implements BlockingStream<TrackedEventMessage<?>> {
```

Comment:
Even though it's private, a little JavaDoc for us developers might be nice (even though I understand the use case of this class).